test_scripts/API/Restructuring_OnResetTimeout/006_BC_OnResetTimeout_with_wrong_methodName_and_HMI_did_not_respond.lua

```diff
@@ -0,0 +1,62 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md
+-- User story: TBD
+-- Use case: TBD
+--
+-- Requirement summary: TBD
+--
+-- Description:
+-- In case:
+-- 1) RPC is requested
+-- 2) Some time after receiving RPC request on HMI is passed
+-- 3) HMI sends BC.OnResetTimeout(resetPeriod = 6000) with right requestID but wrong methodName to SDL
+-- 4) HMI does not send response in 10 seconds after receiving request
+-- SDL does:
+-- 1) Respond in 10 seconds after receiving request
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout')
+local commonRC = require('test_scripts/RC/commonRC')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local wrongMethodName = "Wrong_methodName"
+
+local paramsForRespFunction = {
+	notificationTime = 6000,
+	resetPeriod = 6000
+}
+
+function invaliParamOnResetTimeout(pData, pParams)
+  local function sendOnResetTimeout()
+    common.onResetTimeoutNotification(pData.id, wrongMethodName, pParams.resetPeriod)
+  end
+  RUN_AFTER(sendOnResetTimeout, pParams.notificationTime)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("App_1 registration", common.registerAppWOPTU)
+runner.Step("App_2 registration", common.registerAppWOPTU, { 2 })
+runner.Step("App_2 activation", common.activateApp, { 2 })
+runner.Step("App_1 activation", common.activateApp)
+runner.Step("Set RA mode: ASK_DRIVER", commonRC.defineRAMode, { true, "ASK_DRIVER" })
+runner.Step("Create InteractionChoiceSet", common.createInteractionChoiceSet)
+
+runner.Title("Test")
+for _, rpc in pairs(common.rpcs) do
+  runner.Step("Send " .. rpc , common[rpc],
+    { 11000, 4000, invaliParamOnResetTimeout, paramsForRespFunction, common.rpcResponse })
+end
+
+runner.Step("App_2 registration", common.registerAppWOPTU, { 2 })
```

Comment:
@HSavynetska  app 2 already registered at the line 45

Reply:
@LitvinenkoIra 
Fixed in commit: 1865e46725b3a76d09cde87e801fd27c1979ff8d